app/src/test/java/com/jalasoft/routesapp/util/algorithm/RouteCalculatorTest.kt

```diff
@@ -0,0 +1,69 @@
+package com.jalasoft.routesapp.util.algorithm
+
+import com.jalasoft.routesapp.data.model.remote.LinePath
+import com.jalasoft.routesapp.data.source.RouteAlgorithmFakeData
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import dagger.hilt.android.testing.HiltTestApplication
+import junit.framework.TestCase
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@ExperimentalCoroutinesApi
+@RunWith(RobolectricTestRunner::class)
+@Config(application = HiltTestApplication::class, manifest = Config.NONE)
+@HiltAndroidTest
+class RouteCalculatorTest : TestCase() {
+    @get:Rule
+    val hiltRule = HiltAndroidRule(this)
+
+    private lateinit var lines: List<LinePath>
+
+    @Before
+    public override fun setUp() {
+        super.setUp()
+        val start1 = RouteAlgorithmFakeData.coordinatesToLocation(-16.52035351419114, -68.12580890707301)
+        val points1 = RouteAlgorithmFakeData.arrayToMutableListOfLocation(RouteAlgorithmFakeData.points1Array)
+        val stops1 = RouteAlgorithmFakeData.arrayToMutableListOfLocation(RouteAlgorithmFakeData.stops1Array)
+        val line1 = LinePath("Line 1", "Bus", points1, start1, stops1)
+
+        val start2 = RouteAlgorithmFakeData.coordinatesToLocation(-16.5255314, -68.1254204)
+        val points2 = RouteAlgorithmFakeData.arrayToMutableListOfLocation(RouteAlgorithmFakeData.points2Array)
+        val stops2 = RouteAlgorithmFakeData.arrayToMutableListOfLocation(RouteAlgorithmFakeData.stops2Array)
+
+        val line2 = LinePath("246", "Mini", points2, start2, stops2)
+        hiltRule.inject()
+        lines = listOf(line1, line2)
+    }
+
+    @Test
+    fun `Given a list of lines and an origin an destination point when the line stops are near the given points then it returns a list of single lines`() {
```

Comment:
You're also missing unit test cases here, please add a case where you got 2 points but no routes between them and another case that adds mixed line results (not only single or double, a mixed list between single and double routes)

Reply:
Added two new unit test